# ML-02 — Research Question and Provisional Lane

[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/AdityaPrakash-Kaizu07/Flyrank-AI-intern/blob/main/work/notebooks/w01_research_question.ipynb?flush_cache=true)

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

# Selected Lane: Lane 2 - Refresh / Content Opportunity Scoring

I chose this lane because it solves a direct resource-allocation problem. Content teams cannot update every page. By building a ranked queue, we move beyond predicting abstract metrics and directly influence operational efficiency. Instead of a black-box prediction, this lane requires generating actionable "reason codes" alongside a priority score, making the model transparent and highly practical for an editorial team.

In [10]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


## 2. The question: decision, action, cost of a wrong call

The Search Question: Which content pages should be prioritized for refresh, expansion, or pruning to maximize engagement and ROI?

**Unit of Analysis**: A single content page (URL).
The Output: A ranked queue featuring a suggested action, a priority score, and reason codes per page.

**The Decision & Action:**
Decision: Optimizes how the editorial team allocates limited hours across a large content catalog.

**Action**: Editors work down the ranked queue to execute the suggested updates.

**Cost of a Wrong Call:**
False Positive (Updating a low-impact page): Wasted editorial hours and opportunity cost.

**False Negative** (Ignoring a high-value, decaying page): Lost organic traffic and degrading search visibility.

**Why ML Helps**: Simple rules (e.g., "update pages over a year old") miss real-world context. ML combines multiple signals—like age, CTR drops, and demand—to prioritize updates based on actual opportunity rather than rigid heuristics.

In [11]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [12]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.
import pandas as pd

# Load the starter dataset
df = pd.read_csv('/content/Flyrank-AI-intern/data/raw/content_refresh_anonymized.csv')

# 1. Total scale of the catalog
total_pages = df['content_id'].nunique()

# 2. Stale Content: Pages not updated in over a year (365 days)
stale_pages = len(df[df['days_since_last_update'] > 365])
stale_pct = (stale_pages / total_pages) * 100

# 3. Active Decay: Pages where traffic is currently trending downward
decaying_pages = len(df[df['trend_pct'] < 0])
decaying_pct = (decaying_pages / total_pages) * 100

print(f"Total pages: {total_pages:,}")
print(f"Stale pages (untouched > 1 yr): {stale_pages:,} ({stale_pct:.1f}%)")
print(f"Pages with declining traffic: {decaying_pages:,} ({decaying_pct:.1f}%)")

Total pages: 30,000
Stale pages (untouched > 1 yr): 5 (0.0%)
Pages with declining traffic: 19,715 (65.7%)


## 4. Careful words: what I can and can't claim

# **What I CAN claim:**

 I can claim that this model acts as a robust decision-support tool based strictly on observed historical data (such as CTR, search volume, and engagement decay). The priority scores and reason codes it generates provide directional signals of opportunity, allowing the editorial team to efficiently triage the 19,715 declining pages based on past performance patterns.


# **What I CANNOT claim:**

 I cannot provide causal proof that updating a highly ranked page will mathematically guarantee a traffic reversal. Furthermore, this model is not 'predicting Google'; it cannot foresee external search engine algorithm updates, shifting user intent, or competitor actions. The output is a prioritized queue based on internal dataset metrics, not a crystal ball for search engine rankings.

In [13]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


## Self-check

- [x] **Picked a lane:** Selected predefined Lane 2 (Refresh / Content Opportunity Scoring).
- [x] **Named the decision and action:** Defined the decision (allocating limited editorial hours) and the action (executing updates based on a ranked triage queue).
- [x] **Included real numbers:** Validated the problem scale using code (30,000 total pages, with 19,715 showing declining traffic).
- [x] **Beyond "train a model":** Focused on business ROI, resource allocation, and generating transparent reason codes rather than just optimizing an abstract accuracy metric.
- [x] **Used careful language:** Explicitly stated the model provides directional decision-support and does not offer causal proof or claim to 'predict Google'.